Refactor: dataloader, Add: unittest for dataloader

nayuta723 opened the pull request:
## Issue number
#2 

## Implementation Summary
I added some files with the following file structure.
```
data -- dataset
     |- transform.py
     |- get_dataloader.py
     |- get_dataset.py

test -- test_dataloader.py
```
## Scope of Impact
You can start the training by using the below command
```
$ python train.py
```
## Test
I test the files under "data" folder.
```
$ pytest data/test_dataloader.py
```
## Schedule
4/22

Commit:
Refactor: dataloader, Add: unittest for dataloader

I refactored the function of dataloader and added the unittest for
dataloader.
I added the below file.
data/get_dataloader.py
data/get_dataset.py
data/transform.py
test/test_data;pader

issue #2

tkkawa commented on data/get_dataset.py:
```diff
@@ -0,0 +1,55 @@
+import albumentations as album
+import numpy as np
+import pandas as pd
+import cv2
```
Fix import order.
You should use linter and formatter.

tkkawa commented on train.py:
```diff
@@ -33,38 +33,12 @@
-        
-        return os.path.basename(img_file), img, perf
-        
+# from VGG16_GAP_M2_C32 import model        
```
Delete comment

ryuji0123 commented on data/get_dataset.py:
```diff
@@ -0,0 +1,55 @@
+        img = cv2.imread(img_file, 0)
+        perf = np.nanmedian(data[data['name']==os.path.basename(img_file)[:10]]['Resist']).astype(np.float32)
+
+        if 'T' in os.path.basename(img_file):
```
Unclear to reviewers

ryuji0123 commented on data/transform.py:
```diff
@@ -0,0 +1,25 @@
+
+
+def train_transform() -> album.Compose:
+    """A functions for Data Augmentation on train data
```
Function to transform train data

ryuji0123 commented on data/get_dataloader.py:
```diff
@@ -0,0 +1,42 @@
+
+
+def train_dataloader(files: List[str], csv_file: pd.DataFrame) -> DataLoader:
+    """ A function for train dataloader
```
meaningless comment

ryuji0123 commented on data/get_dataloader.py:
```diff
@@ -0,0 +1,42 @@
+from data.transform import train_transform, val_transform
+
+
+def train_dataloader(files: List[str], csv_file: pd.DataFrame) -> DataLoader:
```
get_train_dataloader

ryuji0123 replied:
files: List[str], csv_file: pd.DataFrame <- very confusing names

ryuji0123 commented on data/get_dataloader.py:
```diff
@@ -0,0 +1,42 @@
+    Returns:
+        DataLoader: train dataloader
+    """
+    args = parse_yacs()
```
Stop parsing here.

ryuji0123 commented on data/get_dataloader.py:
```diff
@@ -0,0 +1,42 @@
+    data = ImageDataset(files=files, csv_file=csv_file, transform=train_transform())
+    return torch.utils.data.DataLoader(
+       data,
+       batch_size=args.TRAIN.BATCH_SIZE,
```
Very difficult to test.

ryuji0123 commented on config/const.py:
```diff
@@ -1,13 +1,17 @@
-print(PROJECT_ROOT)
-TIMESTAMP = "%Y-%m-%d_%H-%M-%S"
+
+TIMESTAMP = str(datetime.datetime.now())
```
Is this correct? In the previous version, TIMESTAMP means time-format. Now it means the loaded time. In addition, I don't think we should use the current TIMESTAMP globally. It doesn't show the correct timestamp.

ryuji0123 commented on config/const.py:
```diff
@@ -1,13 +1,17 @@
 from os import sep
 from os.path import join, dirname, realpath
```
Use pathlib instead.

nayuta723 replied:
Using pathlib would have resulted in code flaws, so I left it as a Join, which is a String object.

ryuji0123 commented on config/const.py:
```diff
@@ -1,13 +1,17 @@
 
 TMP_RESULTS_DIR = join(PROJECT_ROOT, "result", TIMESTAMP)
 ARGS_FILE = join(TMP_RESULTS_DIR, "args.yaml")
 TRAIN_LOG_FILE = join(TMP_RESULTS_DIR, "log.txt")
```
Is this variable necessary in this PR?

ryuji0123 commented on test/test_dataloader.py:
```diff
@@ -0,0 +1,107 @@
+    assert type(transform) == album.Compose, \
+    f"The type of transform should be DataLoader, but {type(transform)}"
+
+def test_ImageDataset():
```
Who told you this code style? It apparently doesn't follow PEP8.

ryuji0123 commented on test/test_dataloader.py:
```diff
@@ -0,0 +1,107 @@
+    dataloader = train_dataloader(files=files, csv_file=csv_file)
+    
+    assert type(dataloader) == DataLoader, \
+    f"The type of data should be DataLoader, but {type(dataloader)}"
```
Why not check dataloader correctly generates batch?

ryuji0123 reviewed (changes requested):
- Delete unnecessary codes.
- Rename very confusing variable and function names.
- Avoid magic numbers.

Commit:
Modify: I modified some code.

I modified data/transform.py, data/get_dataset.py,
data/get_dataloader.py, and test/test_dataloader.py.

issue #3

Commit:
Modify: I modified regarding file paths.

I modified config/const.py and train.py
